Cicedev 254

Renzo-Guerra opened the pull request:
Implementacion de mappers para hacer mas legible el codigo, refactor de algunas implementaciones para poder hacer un correcto manejo de errores desde el frontend. Se creó un manejador de errores para las validaciones del paquete jakarta

Renzo-Guerra committed:
Devolver errores en formato JSON

Cuando se procesa la promesa de hacer un fetch al backend, si lo que
devuelve el back es una cadena de texto, al intentar usar
response.json() en el front se rompe todo y hace mucho mas complicado el
manejo de errores, en cambio devolvuendo un json desde el back aunque
sea un error podemos verificar la peticion con un response.ok

Renzo-Guerra committed:
Refactor del POST /emprendedores

Manejo de la excepcion Unauthorize utilizando el ExceptionHandle

Renzo-Guerra committed:
Se agregaron 2 mappers nuevos para entrepreneur

Renzo-Guerra committed:
ADD excepcion al validar que el id existe

Renzo-Guerra committed:
Validaciones en el DTOEntrepreneurInsert

Renzo-Guerra committed:
@Valid error handler

Si se detectan errores en el @Valid se llama al eventHandler el cual
enviará un objeto con "clave" siendo el "field" y "valor" siendo el "error
message"

Renzo-Guerra committed:
Sacar validacion en el telefono

ciceunicen reviewed (changes requested):
Esta todo muy bueno. Lo único que faltaría seria que una vez que aparece el cartel se redirija al dashboard

Renzo-Guerra committed:
Refactor: Delegar validaciones a userServiceImp

Renzo-Guerra committed:
Solo se puede enviar 1 peticion por userID

Al intentar enviar un formulario para ser emprendedor, solo se puede
enviar 1 peticion por usuario registrado en el sistema
